gslib/commands/du.py

```diff
@@ -113,6 +115,21 @@
       gsutil -o GSUtil:default_project_id=project-name du -shc
 """)
 
+LOCAL_ENC = locale.getpreferredencoding(False)
+
+
+def _GetPatternExclusion(file, position, encoding):
+  exclusion_patterns = []
+  file.seek(position)
+  for line in file:
+    if six.PY2:
+      line = line.strip().decode(encoding)
```

Comment:
Why are we getting unicode values for PY2 (decoding to unicode) and bytes values for PY3 (encoding to some encoding)?

Reply:
The logic was copy/pasted and I didn't dig too far into it. Upon revisiting, a simpler process using `six.ensure` can be used. Changes in #674